src/Graph.ts

```diff
@@ -137,7 +137,10 @@ export default class Graph {
 
 		this.pluginDriver = createPluginDriver(this, options, this.pluginCache, watcher);
 
-		if (watcher) watcher.on('change', id => this.pluginDriver.hookSeqSync('watchChange', [id]));
+		if (watcher) {
+			watcher.removeAllListeners('change');
```

Comment:
Is the `Graph` instance guaranteed to be the only `change` listener?

Reply:
I changed it to move the `removeAllListeners` into the watcher itself exactly for the concerns you mentioned.

Reply:
Please have another look